cmake/ProjectOptions.cmake

```diff
@@ -137,4 +139,13 @@ macro(fiction_local_options)
                              ${ENABLE_UBSAN_MINIMAL_RUNTIME})
   endif()
 
+  # This applies a memory optimization for debug builds which may be used to conform to memory limitations
+  if (FICTION_ENABLE_LIGHTWEIGHT_DEBUG_BUILDS)
+    if (CMAKE_CXX_COMPILER_ID STREQUAL "MSVC")
+      set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} /Z7 /Ob0")
+    else ()
+      set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -g1 -fno-inline")
```

Comment:
Instead of using `else()`, rather check for GNU/GCC and CLANG here. In another `else()` you could then throw a warning regarding an unrecognized compiler.